dtreeviz/trees.py

```diff
@@ -766,6 +778,10 @@ def get_leaves():
         highlight_path = [n.id for n in path]
 
     n_classes = shadow_tree.nclasses()
+
+    # call the get_hex_colors function for up to 40 classes
+    colors['classes'] = get_hex_colors(n_classes,cmap) 
```

Comment:
hi! I can't tell if this defaults to color blind reasonable colors. Does it keep same color scheme for <= 10 classes?

Reply:
This will do `n_classes` discrete colors  evenly spaced from the specified matplotlib colormap which doesn't necessarily mean that they will be colorblind friendly. Ultimately it would be on the user to choose a colormap that is colorblind friendly from something like the [perceptually uniform](https://matplotlib.org/3.5.0/tutorials/colors/colormaps.html#sequential) list of colors. Right now I have it set to default to the `RdYlBu` colormap because that seemed the closest matplotlib colormap to what the original settings were. 

Perhaps a conditional statement for if `n_classes` is less than or equal to 10 that would default to the originally specified colors would work well here?

Reply:
This would change the basic color set then for every user I'd say.  Yeah, we need some thing that's backwards compatible. Maybe the conditional as you say would be enough.

Reply:
I checked here how the viz is looking if number of classes > 10.
<img width="1154" alt="Screenshot 2022-06-13 at 10 33 22" src="https://user-images.githubusercontent.com/12815158/173302632-6fe1b1b8-f8a4-4d15-bc9f-11341244ba50.png">

I agree that we should keep backward compatibilities if n_classes <= 10. But if the n_classes > 10, we can implement add @jlubbersgeo solution. Better to render the dtreeviz than throwing some errors.  https://github.com/parrt/dtreeviz/issues/26 it's another issue complaining about this.

Regarding NUM_BINS limit of 40, we can fix it also. get_num_bins() is just a way to establish the number of bins for a histogram. We can see from the code that the latest values from the list are all of the same value "5". We can choose the same value "5" in case the n_classes > 40.

Reply:
Sounds good. @jlubbersgeo can you add the `n_classes <= 10` check?

Reply:
@parrt yep, I'll take care of it early this week some time!

Reply:
Sounds great!

Reply:
@parrt Sorry for the delay...some work/covid things came up, but I added a line to make everything <= 10 classes keep the original colorblind friendly colors.

Reply:
Hi @jlubbersgeo, thanks and hope you are fine now :)

I checked your code and it works ok, but I observed a little issues (don't figure it out why)

Can you please check the following scenario ?
Try to print a tree with 2 classes, after that with classes > 10 and after that again with 2 classes.
In my case, when you print the tree second time with 2 classes, it doesn't reset the colors to the original, and takes the colors from the tree with classes > 10. Don't understand why :)

Reply:
weird! Let me know if there's any progress and I can try myself and then merge...

Reply:
Looks like `COLORS["colors"]` was being permanently changed after any tree with > 10 classes was being created. Any tree after that even with < 10 classes was then drawing from that list of colors and not the original colorblind friendly color list. It should be fixed such that this doesn't happen. The following test produces the desired results:
```python
from dtreeviz.trees import *
from sklearn.datasets import make_classification
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier

n_classes = [2, 10, 15, 5, 20]
for n in n_classes:
    X, y = make_classification(n_samples=1000, n_classes=n, n_informative=6)
    X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42)
    clf = RandomForestClassifier(max_depth=3)
    clf.fit(X, y)
    tree_num = 20
    viz = dtreeviz(clf.estimators_[tree_num], X, y,cmap = 'viridis')
    viz.view()
```

Reply:
@tlapusan do you have time to check? thanks!

Reply:
@tlapusan great news! Just building on an already great package. Looking forward to including some nice dtreeviz figures in my upcoming presentations/papers.

Reply:
@jlubbersgeo sounds great ! Let us know if you are gonna publish them!

Reply:
now it is pretty clear why the colors classes are overwritten. It's because the colors is a dictionary variable (like static in java), which has a general state and don't reset after each dtreeviz method call.

I'm working on a solution. Also I would like to move the logic of determining the colors values(that if n_classes > 10) inside the colors.py module. It's more appropiate to be there,  otherwise each visualisation method from the library would need to add that if logic.